File: app/src/main/java/com/example/loren/minesample/TranslationActivity.java
package com.example.loren.minesample;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.Toast;

import com.google.gson.Gson;

import bean.TranslationBean;
import butterknife.BindView;
import butterknife.ButterKnife;
import util.HttpUtil;

public class TranslationActivity extends AppCompatActivity {

    @BindView(R.id.input_edt)
    EditText inputEdt;
    @BindView(R.id.output_edt)
    EditText outputEdt;
    @BindView(R.id.translation_iv)
    ImageView translationIv;
    @BindView(R.id.container_ll)
    LinearLayout containerLl;
    private String TRANSLATION_URL = "http://fanyi.youdao.com/openapi.do?keyfrom=Skykai521&key=977124034&type=data&doctype=json&version=1.1&q=";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_translation);
        ButterKnife.bind(this);
        translationIv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                toTranslation();
            }
        });
        containerLl.getChildAt(0).setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                translate();
            }
        });
    }

    private void translate() {

    }


    private void toTranslation() {
        HttpUtil.get(TRANSLATION_URL + inputEdt.getText().toString().trim(), new HttpUtil.NetCallBack() {
            @Override
            public void onSuccess(String json) {
                try {
                    TranslationBean bean = new Gson().fromJson(json, TranslationBean.class);
                    String result = "";
                    for (int i = 0; i < bean.getBasic().getExplains().size(); i++) {
                        result += bean.getBasic().getExplains().get(i) + "  ";
                    }
                    outputEdt.setText(result);
                } catch (Exception e) {
                    e.printStackTrace();
                    Toast.makeText(TranslationActivity.this, "异常了哈哈哈哈哈", Toast.LENGTH_LONG).show();
                }
            }

            @Override
            public void onFailure(String reason) {
                Toast.makeText(TranslationActivity.this, reason, Toast.LENGTH_LONG).show();
            }
        });
    }
}

File: app/src/main/java/com/example/loren/minesample/TranslationActivity.kt
package com.example.loren.minesample

import android.view.View
import bean.TranslationBean
import com.example.loren.minesample.base.ui.BaseActivity
import com.example.loren.minesample.util.http
import com.google.gson.Gson
import kotlinx.android.synthetic.main.activity_translation.*
import pers.victor.ext.toast

class TranslationActivity : BaseActivity() {
    override fun initWidgets() {
        translation_iv!!.setOnClickListener { toTranslation() }
        container_ll!!.getChildAt(0).setOnClickListener { translate() }

    }

    override fun setListeners() {
    }

    override fun onWidgetsClick(v: View) {
    }

    override fun bindLayout() = R.layout.activity_translation

    private val TRANSLATION_URL = "http://fanyi.youdao.com/openapi.do?keyfrom=Skykai521&key=977124034&type=data&doctype=json&version=1.1&q="

    private fun translate() {

    }


    private fun toTranslation() {
        http {
            url = TRANSLATION_URL + input_edt!!.text.toString().trim { it <= ' ' }
            success {
                try {
                    val bean = Gson().fromJson(it, TranslationBean::class.java)
                    var result = ""
                    for (i in 0 until bean.basic.explains.size) {
                        result += bean.basic.explains[i] + "; "
                    }
                    output_edt!!.setText(result)
                } catch (e: Exception) {
                    e.printStackTrace()
                    toast("异常了哈哈哈哈哈")
                }
            }
            fail {
                toast(it)
            }
        }
    }
}
